transformer_engine/common/common.h

```diff
@@ -101,6 +101,7 @@ struct SimpleTensor {
     }
     return acc;
   }
+  bool has_data() const noexcept { return dptr != nullptr; }
```

Comment:
When doing a group GEMM with a split of size zero, we can encounter an edge case where a tensor has a non-null dptr but dptr is not safe to access. See https://github.com/NVIDIA/TransformerEngine/pull/2330 for an attempt to make the behavior of `has_data` consistent. 
```suggestion
  bool has_data() const noexcept { return dptr != nullptr && numel() > 0; }
```